### DISCUSSION TITLE
Multiple day Events shown only once

### DISCUSSION THREAD

**Discussion by ionutcocanu** [MMRIZE/MMM-CalendarExt3Agenda#70] [Q&A] [answered]
Hello,

Would it be possible to display only once (the first date) of a multi-day event? I noticed it is something like that in the default Calendar of the MM and I was wondering if it is achievable as well in the CalendarExt3Agenda module.
![image](https://github.com/user-attachments/assets/1122e6be-484c-4a4d-b4f2-f58db301579c)
I want do display it only once and hide the rest of the days if not other events are planned.

**Comment by eouia** [comment]
But that is the concept of this module. (And I never heard the skippable multiday event option on the default calendar module. what is the option?)
Anyway you can modify the event data itself with `eventPayload` callback function in the config. But... I cannot agree with your intention is useful. What is the benefit of that feature?

**Reply by ionutcocanu** [reply]
Thank you for your answer.

So benefit is that I am using the week mode for MMM-CalendarExt3 on top_bar which display the next 2 weeks (I call it detail version of my calendar since includes normal+sports+other events). And In addition to this, under the Clock I have the MMM-CalendarExt3Agenda which displays only the main events (not the occasional ones) and is more like seeing more into the future of main events for next 1-4 months. 

The issue is the space, as I have for instance 5 days training or 3 days ski trip or 2 days party beeing displayed 5, 3 and 2 times. No problem to show that length on the top_bar module but on the Agenda I wanted to aim for just one line, leave more room for other events.

![image](https://github.com/user-attachments/assets/b0d8b28b-3208-4511-80cc-8dc7d9ee7c2c)

edit: I haven't seen it as an option in the default calendar but is there, it displays only once. Maybe is related to "duplicate" events. Not sure. But if I enable the default calendar, is only once.

**Comment by eouia** [comment, marked as answer]
Add `eventPayload` to your CX3A config.
```js
/* config/config.js MMM-CalendarExt3Agenda config */
eventPayload: (payload) => {
	if (!Array.isArray(payload) || payload.length === 0) return payload
	for (const ev of payload) {
		if (ev.fullDayEvent === true) {
			const newStartDate = new Date(+ev.startDate)
			ev.startDate = String(new Date(newStartDate).setHours(0, 0, 0, 0))
			ev.endDate = String(new Date(newStartDate).setHours(23, 59, 59, 999))
		}
	}
	return payload
},
```
It will show;
**BEFORE**
<img width="625" alt="image" src="https://github.com/user-attachments/assets/9fcd32d9-b2a3-4f95-85cb-b02af38b1383">

**AFTER**
<img width="625" alt="image" src="https://github.com/user-attachments/assets/63df3059-fee5-4da6-8ae2-265ff1d34263">

**Reply by ionutcocanu** [reply]
Thank you very much, works perfectly :)

**Comment by thestumonkey** [comment]
I'm using this method to do the same with a multiday event (example: start 21:00, ends 02:00 the next day).  This just rewrites the end time though so it shows it as finishing at 23:59.  I still want to show the correct end time, I just don't need it on the agenda as an item.
